shared/src/components/buttons/async-button.cjsx

```diff
@@ -40,7 +40,7 @@ module.exports = React.createClass
 
   checkForTimeout: ->
     {isWaiting} = @props
-    @setState(isTimedout: true, delayTimeout: null) if isWaiting and @isMounted()
+    @setState(isTimedout: true, delayTimeout: null) if isWaiting
```

Comment:
No, calling `setState` on an unmounted component doesn't cause an exception, just a warning.  

However Dante points out that we aren't clearing the timeout, but we do have a method to do so. I've added a call to it in `componentWillUnmount`